File: dcp-solutions/2024-12-05/solution.ex
```elixir
defmodule ParenthesesRemover do
  @spec remove_to_make_valid(String.t()) :: non_neg_integer
  def remove_to_make_valid(s) do
    stack = []
    count = 0

    for c in s do
      case c do
       ?(
        when is_integer(n) -> n + 1
      end
    end
  end
end
```

File: dcp-solutions/2024-12-05/solution.go
```markdown
package main

import (
	"fmt"
)

func minRemoveToMakeValid(s string) int {
	stack := []byte{}
	count := 0
	for _, r := range s {
		if r == '(' {
			stack = append(stack, r)
		} else if len(stack) > 0 && stack[len(stack)-1] == '(' {
			stack = stack[:len(stack)-1]
		} else {
			count++
		}
	}

	return count + len(stack)
}
```